src/ngx_http_lua_socket_tcp.c

```diff
@@ -1695,6 +1701,100 @@ ngx_http_lua_socket_tcp_sslhandshake(lua_State *L)
 #endif
                     }
                 }
+
+                if (n >= 6) {
+                    if (lua_type(L, 6) == LUA_TTABLE) {
+                        lua_getfield(L, 6, "client_cert");
+
+                        if (!lua_isnil(L, -1)) {
+#ifdef OPENRESTY_LUAJIT
+                            chain = luaL_checkcdataptr(L, -1);
```

Comment:
I think maybe it's the time to rewrite this `sslhandshake` method with FFI.

Reply:
Oh, if we introduce a separate method to set the client certificates and keys, then we don't need to touch this method at all and just need to introduce a new FFI-based method instead.

Reply:
Yes if we are decided on adding a brand new API then I'm all in for implementing it using FFI.